tests/vmi_multus_test.go

```diff
@@ -995,6 +1004,50 @@ var _ = Describe("[Serial]Macvtap", func() {
 			})
 		})
 	})
+
+	Context("VMI migration", func() {
+		var macvtapVMI *v1.VirtualMachineInstance
+
+		cleanupVMI := func(vmi *v1.VirtualMachineInstance) error {
+			err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(macvtapVMI.GetName(), &metav1.DeleteOptions{})
+			err = wait.PollImmediate(time.Second, 2*time.Minute, func() (done bool, err error) {
+				_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(macvtapVMI.GetName(), &metav1.GetOptions{})
+				return err != nil && errors.IsNotFound(err), nil
+			})
+			return err
+		}
+
+		BeforeEach(func() {
+			nodes := tests.GetAllSchedulableNodes(virtClient)
+			Expect(nodes.Items).ToNot(BeEmpty(), "There should be some compute node")
+
+			if len(nodes.Items) < 2 {
+				Skip("Migration tests require at least 2 nodes")
+			}
+
+			if !tests.HasLiveMigration() {
+				Skip("Migration tests require the 'LiveMigration' feature gate")
+			}
+		})
+
+		BeforeEach(func() {
+			macAddress := "02:03:04:05:06:07"
+			macvtapVMI = createCirrosVMIWithMacvtapDefinedMAC(virtClient, macvtapNetworkName, macAddress)
+		})
+
+		AfterEach(func() {
+			Expect(cleanupVMI(macvtapVMI)).To(Succeed(), "cleaning up the macvtap VMI should have occurred")
+		})
+
+		It("should be successful when the VMI MAC address is defined in its spec", func() {
+			By("starting the migration")
```

Comment:
I think it worth adding connectivity check pre and post migration (maybe creating two vms on host A and migrating both to host B).

Reply:
Could we tackle that on a follow-up PR ?

When we schedule a VM to run in a node, we are basically setting its `NodeAffinity`. When we want it to be migrated, we want it to be moved to a different node - but it copies the NodeAffinity from the original pod, which results in an un-schedulable migration. This seems to be by design, and I don't think a VM that you pin to a node can be migrated.

The way I can do migration would require me to just start 2 VMs - wherever they are scheduled - and connect them to the host's network. This way, they will have connectivity - I would need the guest agent installed in them to 'learn' their IP address.

Then I could migrate them, and afterwards ping yet again.

Reply:
I've gone ahead and implemented the second option I present in the previous comment; It would look like https://gist.github.com/maiqueb/c608a6f9d1cad62bc57036dea05b30c8

If you're OK with this approach (or suggest an alternative into 'calculating the IP addresses), I'll add this new commit.

Reply:
I prefer option 1. Option 2 seems risky to me (I mean risk to set an already used address).
Anyway, let do it on a separate PR and won't block this PR on it.

Reply:
Agreed; Will push a follow-up PR where I'll address adding a migration test with traffic.

In it, will also address how to enable the macvtap (and the remaining tests in this file) to run in parallel.